clients/src/test/java/org/apache/kafka/common/security/JaasUtilsTest.java

```diff
@@ -0,0 +1,260 @@
+        // LoginModuleControlFlag.toString() has format "LoginModuleControlFlag: flag"
+        String[] tokens = loginModuleControlFlag.toString().split(" ");
+        return tokens[tokens.length - 1];
+    }
+
+    private String jaasConfigProp(String loginModule, LoginModuleControlFlag controlFlag, Map<String, Object> options) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(loginModule);
+        builder.append(' ');
+        builder.append(controlFlag(controlFlag));
+        for (Map.Entry<String, Object> entry : options.entrySet()) {
+            builder.append(' ');
+            builder.append(entry.getKey());
+            builder.append('=');
+            builder.append(entry.getValue());
+        }
+        builder.append(';');
+        return builder.toString();
+    }
+
+    private void writeConfiguration(LoginType loginType, String jaasConfigProp) throws IOException {
+        FileWriter writer = new FileWriter(jaasConfigFile);
+        writer.write(loginType.contextName() + " { ");
+        writer.write(jaasConfigProp);
+        writer.write("};");
+        writer.close();
+        Configuration.setConfiguration(null);
+    }
+
+    private void testConfiguration(String loginModule, LoginModuleControlFlag controlFlag, Map<String, Object> options) throws Exception {
+        String jaasConfigProp = jaasConfigProp(loginModule, controlFlag, options);
+        testConfiguration(jaasConfigProp, loginModule, controlFlag, options);
+    }
+
+    private void checkEntry(AppConfigurationEntry entry, String loginModule, LoginModuleControlFlag controlFlag, Map<String, ?> options) {
+        assertEquals(loginModule, entry.getLoginModuleName());
+        assertEquals(controlFlag, entry.getControlFlag());
+        assertEquals(options, entry.getOptions());
+    }
+
+    private void testConfiguration(String jaasConfigProp, String loginModule, LoginModuleControlFlag controlFlag, Map<String, Object> options) throws Exception {
+        AppConfigurationEntry dynamicEntry = configurationEntry(LoginType.CLIENT, jaasConfigProp);
+        checkEntry(dynamicEntry, loginModule, controlFlag, options);
+        assertNull("Static configuration updated", Configuration.getConfiguration().getAppConfigurationEntry(LoginType.CLIENT.contextName()));
+
+        writeConfiguration(LoginType.SERVER, jaasConfigProp);
+        AppConfigurationEntry jaasConfEntry = configurationEntry(LoginType.SERVER, null);
+        checkEntry(jaasConfEntry, loginModule, controlFlag, options);
+    }
+
+    private void testInvalidConfiguration(String jaasConfigProp) throws IOException {
+        try {
+            writeConfiguration(LoginType.SERVER, jaasConfigProp);
+            AppConfigurationEntry entry = configurationEntry(LoginType.SERVER, null);
+            fail("Invalid JAAS configuration file didn't throw exception, entry=" + entry);
+        } catch (SecurityException e) {
```

Comment:
The static JAAS config file parser throws `SecurityException` for `Configuration.getConfiguration()` if the config file is invalid.